### PR TITLE
deduplicate_data_wrappers: Support Empty Buffers

### PR DESCRIPTION
**Pull request by kaushikcfd**
The provided regression fails on `main`.

### CODE REVIEW

File: test/test_codegen.py [hunk 1]
```diff
@@ -1592,6 +1592,34 @@ def get_np_input_args():
     np.testing.assert_allclose(pt_result, np_result)
 
 
+def test_zero_size_cl_array_dedup(ctx_factory):
+    # before pytato@0d8b909 this regression would fail as
```
Comment (inducer) [inline test/test_codegen.py:1596]:
```suggestion
    # At pytato@0d8b909 this regression would fail as
```

File: pytato/transform.py [hunk 2]
```diff
@@ -1500,14 +1500,26 @@ def _get_data_dedup_cache_key(ary: DataInterface) -> Hashable:
...
+                        None,
+                        ary.offset,
+                        ary.shape,
+                        ary.strides,
+                        ary.dtype,
+                        )
+            else:
+                assert isinstance(ary.base_data, CLBuffer)
+                return (
+                        ary.base_data.int_ptr,
```
Comment (inducer) [inline pytato/transform.py:1517]: Why not put the conditional into this tuple entry?

File: test/test_codegen.py (hunk 1 above)
Comment (kaushikcfd) [reply test/test_codegen.py]: Thanks!

File: pytato/transform.py (hunk 2 above)
Comment (kaushikcfd) [reply pytato/transform.py]: Yep, better. Thanks!

**Comment by inducer** [comment]
Thx!

**Pull request merged**